internal/datanode/flush_manager.go

```diff
@@ -175,9 +174,31 @@ func (q *orderFlushQueue) enqueueDelFlush(task flushDeleteTask, deltaLogs *DelDa
 // send into injectCh in there is running task
 // or perform injection logic here if there is no injection
 func (q *orderFlushQueue) inject(inject *taskInjection) {
```

Comment:
I'm a little bit confused about taskInjection, maybe all we need is a barrier.

Reply:
The injector(compactor) start inject and find all running flush tasks, and insert a wait mark for each flusher

Compactor goroutine finish compaction, change all the target id of the affected flusher, and notify all the flusher

Reply:
postInjection postInjectionFunc var is not actually necessary and it shouldn't be a part of orderFlushQueue